### ISSUE TITLE
Research if it is feasible to show ban history somewhere for users based on modmail ban message search.

### ISSUE THREAD

**Issue by creesch** [toolbox-team/reddit-moderator-toolbox#351]
I saw that someone made a dedicated extension to display this on all users. I can't however remember where exactly I saw it and it has the potential to be done wrong using up the api ratelimit. 

So I am thinking about using modmail search with a query like this:

`/api/mod/conversations/search?q=subject%3A"banned from participating in r%2Ftoolbox" participant%3Acreesch`

And then possibly show it in the historybutton or modbutton and modmail sidebar. I think that doing it like that will limit the API calls to a reasonable amount. 

We could then show something like: 

`Banned permanently: 2x`
`Banned temporarily: 1x`
`Last ban date: 2019-09-09`

Or something along those lines. 

We could even expand it to include muting.

**Comment by creesch** [comment]
[Found the inspiration again](https://www.reddit.com/r/ModSupport/comments/i5bde9/idea_keep_a_history_of_bans/g0o2neq/)

**Comment by creesch** [comment]
@Geo1088 @SpyTec can I get your thoughts on the following? 

I was deliberating on adding it to the modbutton (ban history tab or something like that) but I figured that it might make sense to also include it in the historybutton overview. Should we only do that when historybutton is called from a mod subreddit and only show bans for that subreddit, or should we always include it and show bans for all subreddits? 

I am inclined to just show all bans but as reddits policy sort of encourages to only judge people based on actions in the sub itself I am not entirely sure. 


On a more practical note, I'll probably end up making a tbAPI function `getModmailBanHistory` that takes a username and an optional subreddit value (probably also a check if the subreddit is enrolled in new modmail). That way we can show the same data in multiple places. I don't think we really need any caching here as the only place where I think we will show it automatically will be modmail itself and I don't foresee ratelimit issues there.

**Comment by Geo1088** [comment]
I'd probably throw it in history button only. Mod button is mostly about performing actions, not just getting data, at least in my eyes. I don't have a strong preference for all bans or just bans for the current sub, though it would probably be more consistent if we went for the latter since that's how usernotes are displayed already.

**Comment by creesch** [comment]
> I'd probably throw it in history button only. Mod button is mostly about performing actions, not just getting data,

Makes sense, my reasoning for it possibly going there is because it is mod actions related and mod button already shows ban data. But as you said it is mostly about performing those actions. 

> though it would probably be more consistent if we went for the latter since that's how usernotes are displayed already.

Well yeah, but historybutton itself shows data for all subreddits so I was more or less thinking about making it consistent there.

**Comment by Geo1088** [comment]
> but historybutton itself shows data for all subreddits so I was more or less thinking about making it consistent there.

Hadn't thought about that. I like that plan.
